test/end-to-end/test_tls_ctl_gateway.ps1

```diff
@@ -0,0 +1,151 @@
+$etcPath = "/hab/etc"
+$configPath = "$etcPath/cli.toml"
+$ctlTlsPath = "/hab/cache/keys/ctl"
+
+Describe "ctl gateway TLS" {
+    BeforeAll {
+        Remove-Item -Force -Recurse -ErrorAction Ignore $configPath
+        New-Item -ItemType directory -Force -Path $etcPath
+        Remove-Item -Force -Recurse -ErrorAction Ignore $ctlTlsPath
+    }
+
+    It "ctl gateway startup fails without having generated a key and certificate" {
+        {
+            Invoke-NativeCommand hab sup run --ctl-server-certificate --ctl-server-key
+        } | Should -Throw
+    }
+
+    It "ctl gateway key and certificate generation" {
+        Invoke-NativeCommand hab sup secret generate-tls --subject-alternative-name localhost
+    }
+
+    It "ctl gateway TLS connection fails without TLS enabled client" {
+        Start-Supervisor -Timeout 45 -SupArgs @( `
```

Comment:
Do these Start-Supervisor commands need to handle the $logfile parameter now?

Reply:
Ah, right you are! Fix incoming.